pkg/controller/upload-controller.go

```diff
@@ -483,6 +482,25 @@ func (r *UploadReconciler) deleteService(namespace, serviceName string) error {
 	return nil
 }
 
+// updateUploadAnnotations updates annotations without overwritting previous conditions or states
+func updateUploadAnnotations(pvc *corev1.PersistentVolumeClaim, anno map[string]string, pod *v1.Pod, isCloneTarget bool) error {
+	// First we check if the clone failed to avoid changing the running condition back to true
+	if isCloneTarget {
+		cloneFailed, ok := pvc.Annotations[AnnCloneSourcePodFailed]
+		if ok && cloneFailed == "true" {
+			return nil
```

Comment:
I don't think we should return here, we should set the annotations as normal, and in the DV controller in the clone section check for both annotations and update the DV condition accordingly. Something like:
if sourcePod running && targetPod running -> DV running condition = true
if sourcePod !running && targetPod running -> DV running condition = false
if sourcePod running && targetPod !running -> DV running condition = false
if !sourcePod running && targetPod !running -> DV running condition = false
And there we can put a message that is a combination of the message from the source and the message from the target.